Multi-lingual support

Corentin-Aulagnet opened the pull request:
I added support for french but tried to do it in a way that it is relatively easy to add other Latin-alphabet-based languages.

It works with a new class Translator, it stores all the translations for the part names (like string, barrel, etc) in a dictionary.

### What did you fix? <!-- provide a description or issue closes statement -->
I tried to fix Multi-Language OCR Support
closes #67 

### Evidence/screenshot/link to line
![image](https://user-images.githubusercontent.com/69643267/175750878-f8691124-baf0-4735-8bb5-6e7d8f020a33.png)
<!-- You can see my fix [on this line](#line-number-1235234) for where the new data exists -->

### Considerations
- It still need some polish, since all displays are not translated but the detection is good
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[Yes]** I think this is not ideal.
- Have I run the linter? **[No]**
- Is is a bug fix, feature request, or enhancement? **[Enhancement]**

Corentin-Aulagnet committed:
First version of multi lingual support. French is fully added and a little bit tested.
A class Translator is added, it stores all the translations for the part names (like string, barrel, etc)

Corentin-Aulagnet committed:
First version of multi lingual support. French is fully added and a little bit tested.
A class Translator is added, it stores all the translations for the part names (like string, barrel, etc)

dimon222 commented:
Interesting approach, but one of the reasons we don't use this kind of mechanics is that we want use translation payloads provided by Warframe.market (you might be able to see it in source draft PR). Is there a reason you don't use same way and instead manually translate the part names?

D1firehail commented:
The [ReloadItems](https://github.com/WFCD/WFinfo/blob/b9bb4ea3af1035bc54bdd35786d0727d4f5c187c/WFInfo/Data.cs#L153) function takes care of fetching the translation data according to the locale setting (which your change to `SettingsWindow.xaml`  should already take care of). [getLocaleNameData](https://github.com/WFCD/WFinfo/blob/b9bb4ea3af1035bc54bdd35786d0727d4f5c187c/WFInfo/Data.cs#L660) accepts the english name and returns it translated according to the WFM translation data for use in language-specific levenshtein distance functions. You can see #231 for a mostly correct implementation and my comment there for what's missing (although note that Master It is currently broken due to in-game UI changes)

Corentin-Aulagnet committed:
Modified the way the translation is done to use the data fetch online on warframe market. It is similar to https://github.com/WFCD/WFinfo/pull/231
MasterIt has not been tested.
Displays are not translated.

Corentin-Aulagnet commented:
I just pushed another version similar to the one from #231. As I said in the commit message, I did not check either Master It or the translation of the displays. 

I will work on understanding how Master It works.

Corentin-Aulagnet committed:
Modifed the MasterIt code to work with Localization and added the french langage. I was'nt able to test it as it does not work at the time of this commit

Corentin-Aulagnet committed:
Merge branch 'master' into feature/fr-support

Corentin-Aulagnet committed:
The equipment now translates to the locale langage with the addition of resources files to translate all hard coded texts.
The UI texts are not translated yet.
Added a status message to inform the user that the locale has been changed
The equiment is redrawn each time it is opened

Corentin-Aulagnet committed:
Corrected a bug where the culture would not initialze properly hence displaying the initialization messages from the wrong locale.

Corentin-Aulagnet commented:
Ok ok, some news from here. I had a bit of trouble with git after the release of 9.5.4
I tried to get the cleanest history I could after the mistakes I made.

The last interesting commits are focused on UX.
The displays in the equipment window are now in the correct language (locale data come from the market_item.json files.
I implemented resources files  called "strings" and "strings.fr" They can be used to translate all hard coded texts such as status messages etc. I am working on a way to translate the UI (all ard coded texts in the .xaml).
If you have any clues on how to do this, I would appreciate it.

I also have, at the time when I write this message modified the output texts of the Snap-It to display the local names of the parts. I will refactor it to work for all displays.

I think this version is production ready, as long as the users speak a bit of english (UI) but I assume they do if they end up downloading this tool. What do you think ?

KuzyXD commented:
I want it to merge!

dimon222 commented:
Conflict, sadly.

dimon222 commented:
Implemented in #413
